Resources/public/js/views/navigation/ez-navigationitemsubtreeview.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-navigationitemsubtreeview', function (Y) {
+    "use strict";
+    /**
+     * Provides a navigation item implementation to point to a subtree
+     *
+     * @module ez-navigationitemsubtreeview
+     */
+    Y.namespace('eZ');
+
+    /**
+     * The navigation item subtree view. It can be used to represent a
+     * navigation item pointing to a location in the location tree.
+     *
+     * @namespace eZ
+     * @class NavigationItemSubtreeView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.NavigationItemSubtreeView = Y.Base.create('navigationItemSubtreeView', Y.eZ.NavigationItemView, [], {
+        _getName: function () {
+            return Y.eZ.NavigationItemView.NAME;
+        },
+
+        /**
+         * Checks whether the navigation item is selected or not based on the
+         * location id in the route.
+         *
+         * @method match
+         * @param {Object} route the matched route object
+         * @return {Boolean}
+         */
+        match: function (route) {
```

Comment:
usually, method names begin with an infinitive verb so I'll change it for `matchRoute`. (`selected` has nothing to do in the name of the method as the `selected` attribute handling is a pure detail of implementation in the default navigation item view and that might be different depending on the item implementation)